nightwatch/globals.js

```diff
@@ -1,15 +1,86 @@
 const LocalTunnel = require('../src/local-tunnel');
 const TestObservability = require('../src/testObservability');
-const {CUSTOM_REPORTER_CALLBACK_TIMEOUT} = require('../src/utils/constants');
+const {CUSTOM_REPORTER_CALLBACK_TIMEOUT, PID_MAPPING_REGEX, IPC_SERVER_NAME, IPC_EVENTS} = require('../src/utils/constants');
 const CrashReporter = require('../src/utils/crashReporter');
 const helper = require('../src/utils/helper');
 const Logger = require('../src/utils/logger');
+const {v4: uuidv4} = require('uuid');
+const ipc = require('node-ipc');
+const path = require('path');
 
 const localTunnel = new LocalTunnel();
 const testObservability = new TestObservability();
 
 const nightwatchRerun = process.env.NIGHTWATCH_RERUN_FAILED;
 const nightwatchRerunFile = process.env.NIGHTWATCH_RERUN_REPORT_FILE;
+const _tests = {};
+const _testCasesData = {};
+
+const pidMapping = (data) => {
+  const {loggingData, pid} = data;
+  _testCasesData[pid] = loggingData.message.replace('TEST-OBSERVABILITY-PID-TESTCASE-MAPPING-', '').slice(1, -1);
+};
+
+
+const uploadTestLog = async (data) => {
+  try {
+    const {loggingData, pid} = data;
+    const uuid = getUUIDFromPID(pid);
+    testObservability.appendTestItemLog(loggingData, uuid);
+  } catch (error) {
+    CrashReporter.uploadCrashReport(error.message, error.stack);
+  }
+};
+
+const getUUIDFromPID = (pid) => {
+  if (_testCasesData[pid] === undefined) {return}
+  const testCaseStartedId = _testCasesData[pid];
+  const testCaseId = _testCasesData[testCaseStartedId].testCaseId;
+  if (_tests[testCaseId] === undefined) {return}
+
+  return _tests[testCaseId].uuid;
+};
```

Comment:
The function looks good overall, but we could take advantage of optional chaining to make it more concise and elegant.

```suggestion
const getUUIDFromPID = (pid) => {
  const testCaseStartedId = _testCasesData[pid];
  const testCaseId = _testCasesData[testCaseStartedId]?.testCaseId;
  return _tests[testCaseId]?.uuid;
};
```